torch/csrc/Module.cpp

```diff
@@ -1067,6 +1068,13 @@ Call this whenever a new thread is created in order to propagate values from
   py_module.def("_set_neg", [](const at::Tensor & x, bool neg) {
     x._set_neg(neg);
   });
+  py_module.def("_set_storage_via_tensor", [](const at::Tensor & x, const at::Tensor & y, int64_t storage_offset, at::IntArrayRef size, at::IntArrayRef stride) {
+    TORCH_CHECK(y.is_contiguous());
+    x.set_(y.storage(), storage_offset, size, stride);
+  });
+  py_module.def("_is_batched", [](const at::Tensor & x) {
+    return at::isBatchedTensor(x);
+  });
```

Comment:
Oh no, what are you using this for? (still making my way through)

Reply:
It's to detect if an input is a batched tensor, and then skip the crossref in that case. I actually think that we should be able to test this case, but I need to detect that something is a batched tensor and then reconstruct the same thing, but meta (but with appropriate storage/strides so I can't just call `to("meta")`). It feels like I don't have enough API surface but I also didn't look that hard.